_posts/2017-07-29-django.md

```diff
@@ -0,0 +1,151 @@
+
+
+### Creating the first app
+
+Once you're done with the basic setup of your project, you can go forward and create any number of apps you want. Because if you want your website to go viral and take over the internet, you need to have multiple apps :P . Type this command to create your first app.
+
+{% highlight shell %}
+$ python3 manage.py startapp appname
+{% endhighlight %}
+
+This will create an app directory called `appname/` within your project. The main file heirarchy will look like this:
+
+```
+mysite/
+    manage.py
+    mysite/
+        __init__.py
+        settings.py
+        urls.py
+        wsgi.py
+    appname/
+        __init__.py
+        admin.py
+        apps.py
+        migrations/
+            __init__.py
+        models.py
+        tests.py
+        urls.py
+        views.py
+```
+This might probably be confusing at this moment. You already have so many python files and I have not explained what each one of them do. Well, hang tight. I'll get there.
+
+Now that you have your it, you need to include this `appname` in the INSTALLED_APPS tuple in `mysite/setting.py`. This lets Django know that you will be working with an additional app that Django did not know about previously. To complete the step, you need to migrate your database. By doing that, you'll be able to create tables in the database.
+
+Django provides an inbuilt database `SQLite`. To initiate it, type the following commands
+
+{% highlight shell %}
+$ python3 manage.py makemigrations        # Prepare SQL queries
+$ python3 manage.py migrate               # Execute them
+{% endhighlight %}
+
+### Django Models
+
+A model is the single, definitive source of information about your data. It contains the essential fields and behaviors of the data you’re storing. Generally, each model maps to a single database table. All the columns in the table can be specified as the properties of the model, for example,
+
+``` python
+from django.db import models
+
+class Person(models.Model):
+    first_name = models.CharField(max_length=30)
+    last_name = models.CharField(max_length=30)
+```
+Equivalent SQL Query
+
+```
+ CREATE TABLE Person (
+    first_name varchar(30),
+    last_name varchar(30),
+    PRIMARY KEY (id), 
+ );
+```
+
+### Django Views
+
+A view is a callable which takes a request and returns a response. This can be more than just a function, and Django provides an example of some classes which can be used as views. These allow you to structure your views and reuse code by harnessing inheritance and mixins. Views can respond in many ways, e.g through JSON data or by rendering whole XML template. 
+ 
+ 
+### Ending
+
+I along with my project partner created two web applications in Django. Here are the links to them.
+
+* [Gymkhana Nominations Portal](http://github.com/aniketp41/Gymkhana-Nominations.git) *
+* [Awesome Library](http://github.com/SummerCamp17/awesome-library.git)
+
+\* Nominated as the **Best Summer Project** in Programming Club.
```

Comment:
And remove the ** please :)

Thanks for the prompt response to the call for posts, however.